### ISSUE TITLE
Cutoff evaluated for bonds

### ISSUE THREAD

**Issue by y1zhou** [torchmd/torchmd#39]
I'm getting different energy estimates in `forces.compute` vs. `gmx energy`, i.e. the `Bond` term from Gromacs and the `bonds` term from running `Epot = forces.compute(system.pos, system.box, system.forces, returnDetails=True)`differ by a small amount. If I remove the following section then the results become comparable:

https://github.com/torchmd/torchmd/blob/cb7ff8723757f7b3e8318374a6508aa77371b3fa/torchmd/forces.py#L108-L116

Should this be removed similar to what was done in https://github.com/torchmd/torchmd/commit/3f2d103523f5c3aad8af9c4af19acdf95b815e94?

**Comment by stefdoerr** [comment]
That's weird. That would mean that your bond is larger than the cutoff which should not be the case for any normal biological system. Unless your units are wrong maybe?

**Comment by y1zhou** [comment]
Hi @stefdoerr, thanks for the quick response! I was playing with a toy system (alanine tetrapeptide) and just wanted to make sure the system is setup correctly. Here's part of the code:

```python
import parmed as pmd
import numpy as np
import torch
from pathlib import Path
from moleculekit.molecule import Molecule

from torchmd.forcefields.forcefield import ForceField
from torchmd.forces import Forces
from torchmd.parameters import Parameters
from torchmd.systems import System

precision = torch.float
device = torch.device("cuda:0")

# Transform GROMACS files to AMBER format
aln_dir = Path("/path/to/alanine")
outdir = aln_dir / "torch-output"
outdir.mkdir(parents=True, exist_ok=True)

gmx = pmd.load_file(str(aln_dir / "gromacs" / "4ala.top"), xyz=str(aln_dir / "gromacs" / "npt_4ala.gro"))
gmx.save(str(outdir / "4ala.prmtop"), format='amber', overwrite=True)
gmx.save(str(outdir / "4ala.crd"), format='charmmcrd', overwrite=True)

# Setup system
mol = Molecule(str(outdir / "4ala.prmtop"))
mol.read(str(outdir / "4ala.crd"))
ff = ForceField.create(mol, str(outdir / "4ala.prmtop"))
parameters = Parameters(ff, mol, precision=precision, device=device)

from torchmd.integrator import maxwell_boltzmann

system = System(mol.numAtoms, nreplicas=1, precision=precision, device=device)
system.set_positions(mol.coords)
system.set_box(mol.box)
system.set_velocities(maxwell_boltzmann(parameters.masses, T=298, replicas=1))

forces = Forces(parameters, cutoff=1.0, rfa=True, switch_dist=1.0, terms=["bonds", "angles", "dihedrals", "impropers", "1-4", "electrostatics", "lj"])
forces.par.mapped_atom_types = forces.par.mapped_atom_types.to(device)

# Evaluate current energy and forces
Epot = forces.compute(system.pos, system.box, system.forces, returnDetails=True)
Epot
```

The `cutoff=1.0` value was taken from the `rcoulomb` and `rvdw` parameters of my gromacs `.mdp` file. I also had to change `constraints = h-bonds` to `constraints = none` in gromacs because otherwise the `Bonds` term would exclude all H-related bonds, and I didn't find a place in `TorchMD` to specify that bond-to-constraint conversion.

Please let me know if there's anything else I could provide that may help debug this issue. Thanks!

**Comment by stefdoerr** [comment]
Cutoff 1 is wrong because you are cutting off all close interactions. It should be usually around 9 A so that it includes all bonded terms and close-range electrostatics.

**Comment by y1zhou** [comment]
Gotcha, so I assume all units in TorchMD are A and not nm? Changing `cutoff` to 10.0 would indeed yield correct `bonds` energy values, but `lj` and `electrostatics` would be wildly off (values in bold):

| Term | Gromacs | Cutoff 1.0 | Cutoff 10.0 |
| ----- | --------- | --------- | ----------- |
| `bonds` | 9.042375717017208 | 8.804295539855957 | 9.042405128479004 |
| `lj`    | 6.783221797323136 | 6.783212661743164 | **595.5953369140625** |
| `electrostatics` | 297.4545889101338 | 297.4552001953125 | **-4009.18115234375** |

**Comment by stefdoerr** [comment]
Yes coordinate units are in Angstrom. Switch dist should also be 7.5A not 1A.
Electrostatics are probably not comparable because Gromacs is using PME I assume. If you can set Gromacs electrostatics to RFA with cutoff and dielectric constant 78.5 you might get similar results.

**Comment by y1zhou** [comment]
> Electrostatics are probably not comparable because Gromacs is using PME I assume. If you can set Gromacs electrostatics to RFA with cutoff and dielectric constant 78.5 you might get similar results.

Indeed Gromacs uses PME. I'll test RFA and get back to you with the results. Thanks a lot for the help!
